packages/tailwindcss/src/compat/default-theme.ts

```diff
@@ -1,10 +1,62 @@
-import { Theme } from '../theme'
-import { createCompatConfig } from './config/create-compat-config'
+import type { NamedUtilityValue } from '../candidate'
+import { segment } from '../utils/segment'
+import colors from './colors'
+import type { UserConfig } from './config/types'
 
-let theme = new Theme()
+function bareValues(fn: (value: NamedUtilityValue) => string | undefined) {
+  return {
+    // Ideally this would be a Symbol but some of the ecosystem assumes object with
+    // string / number keys for example by using `Object.entries()` which means that
+    // the function that handles the bare value would be lost
+    __BARE_VALUE__: fn,
+  }
+}
+
+let bareIntegers = bareValues((value) => {
+  if (!Number.isNaN(Number(value.value))) {
+    return value.value
+  }
+})
+
+let barePercentages = bareValues((value: NamedUtilityValue) => {
+  if (!Number.isNaN(Number(value.value))) {
+    return `${value.value}%`
+  }
+})
+
+let barePixels = bareValues((value: NamedUtilityValue) => {
+  if (!Number.isNaN(Number(value.value))) {
+    return `${value.value}px`
+  }
+})
+
+let bareMilliseconds = bareValues((value: NamedUtilityValue) => {
+  if (!Number.isNaN(Number(value.value))) {
+    return `${value.value}ms`
+  }
+})
+
+let bareDegrees = bareValues((value: NamedUtilityValue) => {
+  if (!Number.isNaN(Number(value.value))) {
+    return `${value.value}deg`
+  }
+})
+
+let bareAspectRatio = bareValues((value) => {
+  if (value.fraction === null) return
+  let [lhs, rhs] = segment(value.fraction, '/')
+  if (!Number.isInteger(Number(lhs)) || !Number.isInteger(Number(rhs))) return
+  return value.fraction
+})
+
+let bareRepeatValues = bareValues((value) => {
+  if (!Number.isNaN(Number(value.value))) {
+    return `repeat(${value.value}, minmax(0, 1fr))`
+  }
+})
 
 export default {
```

Comment:
This is now basically the same as https://github.com/tailwindlabs/tailwindcss/blob/main/stubs/config.full.js with a few bare value helpers spread in to match the V4 behavior